graph-http/src/client.rs

```diff
@@ -58,10 +60,50 @@ impl Debug for ClientConfiguration {
             .field("connect_timeout", &self.connect_timeout)
             .field("https_only", &self.https_only)
             .field("min_tls_version", &self.min_tls_version)
+            .field("proxy", &self.proxy)
             .finish()
     }
 }
 
+/// This is a thin wrapper for [`reqwest::Proxy`].
+#[derive(Debug, Clone)]
+pub struct Proxy {
+    proxy: reqwest::Proxy,
+}
+
+impl Proxy {
+    /// Create proxy by a proxy url.
+    ///
+    /// If not a valid proxy url, return `None`.
+    ///
+    /// # Examples
+    ///
+    /// ```ignore
+    /// assert!(Proxy::new("http://127.0.0.1:3128").is_some());
+    /// assert!(Proxy::new("ftp://192.168.0.1").is_none());
+    ///
+    /// // `socks5` scheme require `socks-proxy` feature.
+    /// assert!(Proxy::new("socks5://myproxyserver.com").is_some());
+    /// ```
+    pub fn new(url: &str) -> Option<Self> {
+        reqwest::Proxy::all(url).ok().map(|proxy| Self { proxy })
+    }
+
+    /// Set the `Proxy-Authorization` header using Basic auth.
+    pub fn basic_auth(self, username: &str, password: &str) -> Self {
+        Self {
+            proxy: self.proxy.basic_auth(username, password),
+        }
+    }
+
+    /// Set the `Proxy-Authorization` header to a specified value.
+    pub fn custom_http_auth(self, header_value: HeaderValue) -> Self {
+        Self {
+            proxy: self.proxy.custom_http_auth(header_value),
+        }
+    }
+}
+
```

Comment:
This isn't really needed. In the GraphClientConfiguration method just use the reqwest proxy as the type.

Reply:
Removed.